### ISSUE TITLE
Feature Request: unify version number for protoc-gen-go-grpc and grpc

### ISSUE THREAD

**Issue by jagobagascon** [grpc/grpc-go#3992]
### Use case(s) - what problem will this feature solve?
`grpc-go` and `prococ-gen-go-grpc` are distributed independently, each with its own version number, and that **makes it hard to know which version of the runtime should be used for a specific generator**.

Other implementation such as `grpc-java` or `grpc-kotlin` keep the same version number for the generator and the runtime, making it much clearer.

This change would make the grpc ecosystem a little more consistent.

### Proposed Solution
Use the same version number for both, `grpc-go` and `protoc-gen-go-grpc`.

### Alternatives Considered
--

### Additional Context
We are currently generating our APIs for several languages. Our process infers the grpc runtime dependencies based on the generator used. For languages other than Go it's just the same version number, but for Go we have to explicitly set a runtime version. So this change would make it a little bit easier for us.

**Comment by dfawley** [comment]
We would rather keep the versions independent for two reasons:

1. Changes to `protoc-gen-go-grpc` happen at a much slower pace than gRPC, and independently.  We may want to release `protoc-gen-go-grpc` without doing a release of the rest of gRPC and vice-versa.

2. the "required gRPC version" of the generated code is even more slowly incremented -- it's very typical that you can use a 1+ year old version of gRPC with the latest version of `protoc-gen-go-grpc`.


Maybe we can find another way to simplify your process -- e.g. would it help if we introduced an exported string to the generated code that contains the minimum required gRPC version?

**Comment by jagobagascon** [comment]
I think that may work for us.

We will maintain both versions manually (not to stay in a very old version of the runtime), but with that change we could just add some sanity-checking to avoid any problem with version incompatibility.

**Comment by dfawley** [comment]
FWIW I could not find any reasonable way to export a string for this, since it's possible multiple files exist with gRPC services.  It's even technically possible those were generated with different versions of `protoc-gen-go-grpc`.  Instead, I added a *comment* indicating when the required `SupportPackageIsVersion#` symbol was added to grpc-go.  I'm not sure there's much else we can do with this, but feel free to suggest other ideas if you have any.
